eppo/src/androidTest/java/cloud/eppo/android/EppoClientTest.java

```diff
@@ -360,13 +356,76 @@ private List<JsonElement> getJSONAssignments(AssignmentTestCase testCase) {
         return (List<JsonElement>) this.getAssignments(testCase, AssignmentValueType.JSON);
     }
 
-    private static String getMockRandomizedAssignmentResponse() {
+    @Test
+    public void testInvalidConfigJSON() {
+
+        // Create a mock instance of EppoHttpClient
+        EppoHttpClient mockHttpClient = mock(EppoHttpClient.class);
+
+        doAnswer(invocation -> {
+            RequestCallback callback = invocation.getArgument(1);
+            callback.onSuccess(new StringReader("{}"));
+            return null; // doAnswer doesn't require a return value
+        }).when(mockHttpClient).get(anyString(), any(RequestCallback.class));
+
+        Field httpClientOverrideField = null;
         try {
-            InputStream in = ApplicationProvider.getApplicationContext().getAssets()
-                    .open("rac-experiments-v3-hashed-keys.json");
-            return IOUtils.toString(in, Charsets.toCharset("UTF8"));
+            // Use reflection to set the httpClientOverride field
+            httpClientOverrideField = EppoClient.class.getDeclaredField("httpClientOverride");
+            httpClientOverrideField.setAccessible(true);
+            httpClientOverrideField.set(null, mockHttpClient);
+
+
+            initClient(TEST_HOST, true, true, false);
+        } catch (NoSuchFieldException | IllegalAccessException e) {
+            throw new RuntimeException(e);
+        } finally {
+            if (httpClientOverrideField != null) {
+                try {
+                    httpClientOverrideField.set(null, null);
+                } catch (IllegalAccessException e) {
+                    throw new RuntimeException(e);
+                }
+                httpClientOverrideField.setAccessible(false);
+            }
+        }
+
+        String result = EppoClient.getInstance().getStringAssignment("dummy subject", "dummy flag");
+        assertNull(result);
+    }
+
+    @Test
+    public void testCachedBadResponseAllowsLaterFetching() {
+        // Populate the cache with a bad response
+        ConfigCacheFile cacheFile = new ConfigCacheFile(ApplicationProvider.getApplicationContext());
+        cacheFile.delete();
+        try {
+            cacheFile.getOutputWriter().write("{}");
+            cacheFile.getOutputWriter().close();
         } catch (IOException e) {
-            throw new RuntimeException("Error reading mock RAC data", e);
+            throw new RuntimeException(e);
         }
+        initClient(TEST_HOST, false, false, false);
+;
+        String result = EppoClient.getInstance().getStringAssignment("dummy subject", "dummy flag");
+        assertNull(result);
+        // Failure callback will have fired from cache read error, but configuration request will still be fired off on init
```

Comment:
Side note: I think we should consider reworking things so that we can use another callback on loadFromCache before firing off our request. The way things are now, if there is a cache file the initialization callback will be used regardless on whether the cache was used or not. So if the cache failed but we got a valid configuration 500ms later, `onFailure()` will be called, which I don't think fully represents what really happened.

Fixing that is outside the scope of this PR, but if we're interested in changing this we can file another ticket.